drivers/modem/ehs6.c

```diff
@@ -0,0 +1,1005 @@
+/*
+ * Copyright (c) 2018 Foundries.io
+ * Copyright (c) 2019 Song Qiang <songqiang1304521@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define LOG_DOMAIN modem_ehs6
+#define LOG_LEVEL CONFIG_MODEM_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(LOG_DOMAIN);
+
+#include <zephyr/types.h>
+#include <stddef.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <ctype.h>
+#include <errno.h>
+#include <zephyr.h>
+#include <gpio.h>
+#include <device.h>
+#include <init.h>
+
+#include <net/net_context.h>
+#include <net/net_if.h>
+#include <net/socket_offload.h>
+#include <net/socket.h>
+#include <net/net_offload.h>
+#include <net/net_pkt.h>
+#if defined(CONFIG_NET_IPV4)
+#include "ipv4.h"
+#endif
+#if defined(CONFIG_NET_UDP)
+#include <net/udp.h>
+#include "udp_internal.h"
+#endif
+#if defined(CONFIG_NET_TCP)
+#include <net/tcp.h>
+#endif
+
+#include <drivers/modem/modem_receiver.h>
+
+/* Uncomment the #define below to enable a hexdump of all incoming
+ * data from the modem receiver
+ */
+/* #define ENABLE_VERBOSE_MODEM_RECV_HEXDUMP	1 */
+
+#define MDM_UART_DEV_NAME		DT_UART_MCUX_2_NAME
```

Comment:
This is currently hard-coded for only the K64F UART 2, is that correct?  Seems like this could be made generic the same was Wistron modem does it by placing the modem node inside the uart device it's attached to.   See:
https://github.com/zephyrproject-rtos/zephyr/blob/master/drivers/modem/wncm14a2a.c#L94

The `DT_WNC_M14A2A_0_BUS_NAME` define is dynamically generated via DTS based on which UART node the WNC modem DTS is placed inside.